### ISSUE TITLE
Include DLL in to Module

### ISSUE THREAD

**Issue by jkdba** [PSKeePass/PSKeePass#2]
Self contained,
Always using the right version.

Todo:

- [x] Embed KeePassLib
- [x] Perform Version Checks
- [x] Consider and Protect against security risks of embedding the KeePassLib

**Comment by jkdba** [comment]
@chritea Do you have more information on the method you mentioned previously?

**Comment by chritea** [comment]
@jkdba nope, not yet. I´m waiting for an answer to get more information

**Comment by jkdba** [comment]
@chritea For the time being I implemented embedding the KeePass Library using the method found [here](http://www.getautomationmachine.com/en/company/news/item/embedding-files-in-powershell-scripts)

**Comment by jkdba** [comment]
@chritea ok so I found a better way to load the embed the Assembly. The new method, stores an compressed base64 encoded string of the assembly, it will then load it if it isnt already loaded, by deflating it and unencoding it back to a byte array then loading it from the byte array to memory.

Found this method [here](http://www.exploit-monday.com/2012/12/in-memory-dll-loading.html)

So from our previous conversation I believe you mentioned that you saw a method where the embedded assembly was encrypted, or something. I would be interested in how that would work, or add any extra security without requiring manual intervention.
